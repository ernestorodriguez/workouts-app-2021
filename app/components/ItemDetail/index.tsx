import React, { ReactElement, useEffect } from "react";
import PropTypes from "prop-types";
import { useSelector, useDispatch } from "react-redux";
import { getItem } from "../../client/actions";
import { RootState } from "../../client/reducers";

interface ItemDetailProps {
  alias: string;
}

const ItemDetail = ({ alias }: ItemDetailProps): ReactElement => {
  const { name, startDate, category, description, thumbnailHigh } = useSelector(
    ({ itemDetail }: RootState) => itemDetail
  );
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getItem(alias));
  }, []);

  return (
    <section className="item-detail">
      <header data-js="name">
        <h2>
          {name}
          <br />
          <small data-js="start-date">{startDate}</small>
        </h2>
      </header>
      <div className="item-detail__badge" data-js="category">
        {category}
      </div>
      <img className="item-image" src={thumbnailHigh} />
      <article data-js="description">{description}</article>
    </section>
  );
};

ItemDetail.propTypes = {
  alias: PropTypes.string.isRequired,
};

export default ItemDetail;
