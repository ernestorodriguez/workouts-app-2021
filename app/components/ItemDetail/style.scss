.item-detail {
  padding: 1em;
  max-width: 500px;
  margin: 0 auto;
}

.item-image {
  width: 100%;
  height: auto;
}

.item-detail__badge {
  background-color: lighten($color: #ff7f66, $amount: 5);
  padding: 5px 10px;
  border-radius: 10px;
  display: inline-block;
  overflow: hidden;
  color: white;
  font-weight: 400;
}
