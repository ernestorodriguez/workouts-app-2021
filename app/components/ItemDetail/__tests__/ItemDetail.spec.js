/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import ItemDetail from '../index';

jest.spyOn(React, 'useEffect').mockImplementation((f) => f());

describe('ItemDetail component', () => {
  test.skip('should match snapshot', () => {
    const itemDetail = {
      name: 'NAME',
      startDate: '2020-10',
      category: 'c1',
      description: 'DESCRIPTION',
    };
    const component = (
      <ItemDetail itemDetail={itemDetail} alias="alias-path" getData={() => {}} />
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
