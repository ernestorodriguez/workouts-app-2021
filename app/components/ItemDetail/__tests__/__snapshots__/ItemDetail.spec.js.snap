// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail component should match snapshot 1`] = `
<section
  className="item-detail"
>
  <header
    data-js="name"
  >
    <h2>
      NAME
      <br />
      <small
        data-js="start-date"
      >
        2020-10
      </small>
    </h2>
  </header>
  <div
    className="item-detail__badge"
    data-js="category"
  >
    c1
  </div>
  <article
    data-js="description"
  >
    DESCRIPTION
  </article>
</section>
`;
